arangod/Aql/Query.cpp

```diff
@@ -1448,3 +1469,58 @@ aql::ExecutionState Query::cleanupTrxAndEngines(ErrorCode errorCode) {
     return ExecutionState::DONE;
   }
 }
+
+  void Query::debugKillQuery() {
+#ifndef ARANGODB_ENABLE_FAILURE_TESTS
+    TRI_ASSERT(false);
+    return;
+#else
+    if (_wasDebugKilled) {
+      return;
+    }
+    bool usingSystemCollection = false;
+    // Ignore queries on System collections, we do not want them to hit failure points
+    collections().visit([&usingSystemCollection](std::string const&, Collection& col) -> bool {
+      if (col.getCollection()->system()) {
+        usingSystemCollection = true;
+        return false;
+      }
+      return true;
+    });
+
+    if (usingSystemCollection) {
+      LOG_DEVEL << "Using system collection: " << std::boolalpha << usingSystemCollection;
+      return;
+    }
+
+    _wasDebugKilled = true;
+    // A query can only be killed under certain circumstances.
+    // We assert here that one of those is true.
+    // a) Query is in the list of current queries, this can be requested by the user and the query can be killed by user
+    // b) Query is in the query-registery. In this case the query registry can hit a timeout, which triggers the kill
+    // c) The query id has been handed out to the user (stream query only)
+    bool isStreaming = queryOptions().stream;
+    bool isInList = false;
+    bool isInRegistry = false;
+    auto const& queryList = vocbase().queryList();
+    if (queryList->enabled()) {
+      auto const& current = queryList->listCurrent();
+      for (auto const& it : current) {
+        if (it.id == _queryId) {
+          isInList = true;
+          break;
+        }
+      }
+    }
+
+    QueryRegistry* registry = QueryRegistryFeature::registry();
+    if (registry != nullptr) {
+      isInRegistry = registry->queryIsRegistered(vocbase().name(), _queryId);
+    }
+    LOG_DEVEL << "Query killed for debugging in current: " << std::boolalpha << isInList << " ,streaming: " << isStreaming << ", in QueryRegistry: " << isInRegistry;
+    LOG_DEVEL << "Query string is: " << queryString();
```

Comment:
Do we need these message? If yes, let's make them proper `LOG_TOPIC(...)` messages.
```suggestion
    LOG_DEVEL << "Query killed for debugging in current: " << std::boolalpha << isInList << ", streaming: " << isStreaming << ", in QueryRegistry: " << isInRegistry;
    LOG_DEVEL << "Query string is: " << queryString();
```

Reply:
No we do not need this message, we used it during development to make sure we actually hit the FailurePoints when we expect them. And stupid me forgot to cleanup properly.